subsys/bluetooth/host/id.c

```diff
@@ -1969,21 +1971,31 @@ int bt_id_set_adv_own_addr(struct bt_le_ext_adv *adv, uint32_t options,
 			 * problem.
 			 */
 #if defined(CONFIG_BT_OBSERVER)
-			bool scan_enabled = false;
+			bool scan_disabled = false;
 
 			/* If active scan with NRPA is ongoing refresh NRPA */
 			if (!IS_ENABLED(CONFIG_BT_PRIVACY) &&
 			    !IS_ENABLED(CONFIG_BT_SCAN_WITH_IDENTITY) &&
 			    atomic_test_bit(bt_dev.flags, BT_DEV_SCANNING)) {
-				scan_enabled = true;
+				scan_disabled = true;
```

Comment:
yikes, this seems like a very old bug :o

Reply:
We should probably conditionally set `scan_disabled` based on the return value of `bt_le_scan_set_enable`

Reply:
have now added this as a separate commit